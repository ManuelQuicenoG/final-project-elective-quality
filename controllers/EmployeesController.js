import axios from 'axios';

export const employeeController = {
    register(employee) {
        return axios({
            method: 'POST',
            baseURL: process.env.API,
            url: `employees`,
            data: employee,
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'employees'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    },
    getemployee(name) {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    },
    updateemployee(name) {
        return axios({
            method: 'PUT',
            baseURL: process.env.API,
            url: `employees/${name}`,
        });
    }
}

