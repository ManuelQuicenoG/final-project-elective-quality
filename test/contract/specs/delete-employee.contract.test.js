import { provider } from '../config/initPact';
import { employeeController } from '../../../controllers';
import { Matchers } from '@pact-foundation/pact';

const nameemployee = 'prueba';

describe('delete an employee service', () => {
    beforeAll(async() => {
        await provider.setup();
    });

    describe('When a request to remove a employee arrived ', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'backend service delete is ready',
                uponReceiving: 'a request to delete a employee',
                withRequest: {
                    method: 'DELETE',
                    path: '/employees/'+nameemployee
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike({
                        message: Matchers.string('Employee deleted successfully'),
                    }, {min: 1})
                }
            });
        });

        test('Then it should return a confirm message that indicate successfully process', async() => {
            const response = await employeeController.delete(nameemployee);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});
