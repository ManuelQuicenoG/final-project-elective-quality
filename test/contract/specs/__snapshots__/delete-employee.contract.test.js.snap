// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete an employee service When a request to remove a employee arrived  Then it should return a confirm message that indicate successfully process 1`] = `
Array [
  Object {
    "message": "Employee deleted successfully",
  },
]
`;
